libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -49,6 +50,11 @@ void StaticAnalyzer::endVisit(ContractDefinition const&)
 bool StaticAnalyzer::visit(FunctionDefinition const& _function)
 {
 	m_nonPayablePublic = _function.isPublic() && !_function.isPayable();
+
+	for (Declaration const* declaration: m_globals)
```

Comment:
@chriseth any idea why here `m_globals.size() == 0`, while in the constructor it is still 18?